rust/automerge/src/iter/spans.rs

```diff
@@ -303,15 +531,9 @@ impl SpanInternal {
         match self {
             SpanInternal::Text(text, _, marks) => Span::Text {
                 text,
-                marks: marks.and_then(|m| {
-                    if m.non_deleted_marks().is_empty() {
```

Comment:
I wonder if we should now delete `MarkSet::non_deleted_marks`, it's not used anywhere else and it's kind of gross.